DOCS-9626 revamped the topic (#365)
mulesoft/docs-api-manager: modules/ROOT/pages/api-proxy-landing-page.adoc

When you configure your API proxy, Anypoint Platform enables you to deploy the proxy application directly to CloudHub or Anypoint Runtime Fabric. Your proxy application is then automatically tracked by API Manager. 

The API proxy is agnostic to your backend's location and programming language. Additionally, your backend can be a non-Mule application.

== When to Use API Proxies

You can use API proxies:

* If your API is live, but not yet hosted in a Mule runtime engine (Mule) server
* If you already have a Mule application, which is closed-code, but don’t have access to include xref:configure-autodiscovery-4-task[autodiscovery] for it
* If you want to perform schema validation on every incoming request for a RAML, OAS, or SOAP API

Depending on the type of specification you used to create your API, you can apply one of the following API proxies:

* HTTP or xref:building-https-proxy.adoc[HTTPS]
* xref:building-soap-proxy.adoc[WSDL]
* xref:proxy-deploy-raml-oas-proxy.adoc[RAML or OAS]

== How API Proxies Work

> You can download a preconfigured Mule application, if you are running an on-premise Mule instances, and deploy it in the same way as any other Mule application, by xref:org-credentials-config-mule4.adoc[configuring your Mule instance with the correct Anypoint Platform credential].

Reviewer: ```suggestion
You can download a preconfigured Mule application, if you are running an on-premises Mule instance, and deploy it in the same way as any other Mule application, by xref:org-credentials-config-mule4.adoc[configuring your Mule instance with the correct Anypoint Platform credential].
```

Author: Done! Thanks!

endif::[]
:keywords: proxy, api, auto-generated, raml, wsdl, apikit, http

> API Proxies function as intermediaries between the external applications and the backend server. When you deploy an API proxy in front of your API, the proxy adopts API gateway capabilities to secure the APIs using different types of xref:policies-landing-page.adoc[policies]. 

Reviewer: ```suggestion
API proxies function as intermediaries between the external applications and the backend server. When you deploy an API proxy in front of your API, the proxy adopts API gateway capabilities to secure the APIs using different types of xref:policies-landing-page.adoc[policies]. 
```

Reviewer: Can you clarify what this means: "deploy an API proxy in front of your API"? Do you mean that you deploy the proxy before you deploy the API, or that the proxy is prepended to the API, or...?

Is the point simply that the proxy uses policies to protect the backend by acting as a gateway between it and the API?

> When you configure your API proxy, Anypoint Platform enables you to deploy the proxy application directly to CloudHub or Runtime Fabric. Your proxy application is then automatically tracked by API Manager. 

Reviewer: ```suggestion
When you configure your API proxy, Anypoint Platform enables you to deploy the proxy application directly to CloudHub or Anypoint Runtime Fabric. Your proxy application is then automatically tracked by API Manager. 
```

Author: Done! Thanks!

> The API proxy is agnostic to where and using which programming language your backend is implemented. Additionally, your backend can be a non-Mule application.

Reviewer: ```suggestion
The API proxy is agnostic to your backend's location and programming language. Additionally, your backend can be a non-Mule application.
```

Author: Done! Thanks!

== When to Use API Proxies

You can use API proxies:

> * If you already have your API live, but is not yet hosted in a Mule runtime engine (Mule) server.

Reviewer: ```suggestion
* If your API is live but not yet hosted in a Mule runtime engine (Mule) server
```

Author: Done! Thanks!

> * If you already have a Mule application, which is closed-code, and don’t have access to include an xref:configure-autodiscovery-4-task[Autodiscovery] for it. 

Reviewer: ```suggestion
* If you already have access to a Mule application with closed code but don’t have xref:configure-autodiscovery-4-task[autodiscovery] access to it
```

Author: Done! Thanks!

> * If you want to perform schema validation on every incoming request for a RAML, OAS, or SOAP API.

Reviewer: ```suggestion
* If you want to perform schema validation on every incoming request for a RAML, OAS, or SOAP API
```

Author: Done! Thanks!

> Depending on the type of specification you used to create for your API, you can apply one of the appropriate API proxies:

Reviewer: ```suggestion
Depending on the type of specification you used to create your API, you can apply one of the following API proxies:
```

Author: Done! Thanks!

* HTTP or xref:building-https-proxy.adoc[HTTPS]
* xref:building-soap-proxy.adoc[WSDL]
* xref:proxy-deploy-raml-oas-proxy.adoc[RAML or OAS]

== How API Proxies Work

> You can download a preconfigured Mule application, if you are running an on-premise Mule instance, and deploy it just as any other Mule application by simply xref:org-credentials-config-mule4.adoc[configuring your Mule instance with the correct Anypoint Platform credential].

Reviewer: GLOBAL: Always use "premises," plural, whether using the adjective form (on-premises) or the adverb form "on premises")

Reviewer: ```suggestion
You can download a preconfigured Mule application, if you are running an on-premise Mule instances, and deploy it in the same way as any other Mule application, by xref:org-credentials-config-mule4.adoc[configuring your Mule instance with the correct Anypoint Platform credential].
```

Author: Done! Thanks!

> The following diagram illustrates the how a request communicates with the proxy endpoint and returns a response from the backend sever:

Reviewer: ```suggestion
The following diagram illustrates how a request communicates with the proxy endpoint and returns a response from the backend server:
```

Author: Done! Thanks!

image::api-gateway-capabilities-mule4-62bf9.png[align=center]
 

> As shown in the diagram, when an external consumer application sends a request, it first pings the proxy endpoint. At this time, all policies applied to the API that are referenced in the proxy application takes effect. 

Reviewer: policies take effect
```suggestion
As shown in the diagram, when an external consumer application sends a request, it first pings the proxy endpoint. At this time, all policies applied to the API and referenced in the proxy application take effect. 
```

Author: Done! Thanks!

> If a policy fails any of the validations, an error response is returned (appropriate to the failed policy) and the request does not reach your backend. Conversely, if all the validations and actions performed by the policies are successful, the request continues to your backend API, and the proxy then returns this response to the consumer application.

Reviewer: "response is returned" is redundant.

Reviewer: "appropriate to the failed policy" is unnecessary because the sentence begins with "if a policy fails."

Reviewer: ```suggestion
If a policy fails any of the validations, the request does not reach your backend API. Conversely, if all the validations are successful, the request continues to your backend API for processing, and the proxy enables the response to return to the consumer application.
```

Reviewer: I wonder if it is misleading to say "a policy fails the validations"? The policy is not what fails, right? The policy is what protects the backend from invalid requests? So, "if the policy discovers an invalid request"?

Author: Done! Thanks!

 
== API Proxy Advantages

> API proxies can improve the performance of your team and secure your APIs by using policies and validations. The API proxies:

Reviewer: More active?
```suggestion
API proxies use policies to validate incoming requests, improving team performance and backend API security. The API proxies:
```

Author: Done! Thanks!

> * Increases Development Time

Reviewer: proxies increase
```suggestion
* Increase development time
```

Reviewer: But how is increasing dev time an advantage? Do you mean "decrease"?

Reviewer: GLOBAL: These are bullet-list items, not subheads, so use sentence-style capitalization.

Author: Done! Thanks!

> * Increases Development Time

Reviewer 2: I saw this comment and made me think about the word I used 🤔 ... I agree that `increases` is not the happier word: My intention was to say that **improves** the development time, or that it **increases** productivity, because it makes things easy for you. So, probably I would use `improves` or change the title entirely to something that reflects that.

Author: Done! Thanks!

+

> You can deploy a proxy directly to a xref:runtime-fabric::index.adoc[Runtime Fabric], xref:runtime-manager::cloudhub.adoc[CloudHub], xref:runtime-manager::deployment-strategies#hybrid-deployments.adoc[Hybrid] or xref:mule-runtime::mule-deployment-model.aodc[standalone Mule runtime engine] instance with just a few clicks. Even if you don't know how to create a Mule Application, API Manager builds and configures the Mule Application with an Autodiscovery so your API can be automatically tracked by API Manager after the deployment has completed.

Reviewer: ```suggestion
You can deploy a proxy directly to a xref:runtime-fabric::index.adoc[Runtime Fabric], xref:runtime-manager::cloudhub.adoc[CloudHub], xref:runtime-manager::deployment-strategies#hybrid-deployments.adoc[Hybrid], or xref:mule-runtime::mule-deployment-model.aodc[standalone Mule] instance with just a few clicks. Even if you don't know how to create a Mule application, API Manager builds and configures it using autodiscovery, so your API can be automatically tracked by API Manager after the deployment has completed.
```

Author: Done! Thanks!

> * Provides Flexibility for Customizations

Reviewer: ```suggestion
* Provides flexibility for customizations
```
OR
```suggestion
* Enables customization
```

Author: Done! Thanks!

+

> In most cases, the proxy generated in API Manager is suitable for deployment. However, because it is a Mule , you can edit it using xref:studio::import-export-packages.adoc[Anypoint Studio] to fulfill additional requirements and customization.

Reviewer: ```suggestion
In most cases, the proxy generated in API Manager is suitable for deployment. However, in the Mule environment, you can use xref:studio::import-export-packages.adoc[Anypoint Studio] to edit the proxy to meet custom requirements.
```

Author: Done! Thanks!

> * Secures and Governs your APIs Using Policies and xref:viewing-api-analytics.adoc[API Analytics]

Reviewer: ```suggestion
* Secures and governs your APIs using policies and xref:viewing-api-analytics.adoc[API analytics]
```
...or do you mean the MuleSoft product, Anypoint Analytics?

Author: Done! Thanks!

+

> The proxy enables you to protect your API with the full capabilities of the API gateway, including access to API Analytics.

Reviewer: See question above: "API analytics" or "Anypoint Analytics"?

Author: API Analytics! Done! Thanks!

> * Implements Validations

Reviewer: ```suggestion
* Implements validations
```

Author: Done! Thanks!

+

> Most proxies, including RAML, REST, and WSDL proxies enable you to perform validations on all incoming requests against your API definition that you design. You can choose different levels of validation, depending on your requirements:

Reviewer: ```suggestion
Most proxies, including RAML, REST, and WSDL proxies, enable you to perform validations on all incoming requests, using your API definition. You can choose different levels of validation, depending on your requirements:
```

Author: Done! Thanks!

** Regular validations
+ 

> Compares the payload, query parameters, URI parameters, headers, and form parameters against the schema defined in your API specification. When using this configuration, the unspecified query parameters and headers in the API specification are also sent to the backend service. 

Reviewer: validations compare; compare with
```suggestion
Compare the payload, query parameters, URI parameters, headers, and form parameters with the schema defined in your API specification. When using this configuration, the unspecified query parameters and headers in the API specification are also sent to the backend service. 
```
Does your audience understand what you mean by "unspecified"? Is it safe to validate an unspecified value?

Author: Yes! Done! Thanks!

> ** Strict Validations 

Reviewer: ```suggestion
** Strict validations 
```

Author: Done! Thanks!

+

> Accepts only those requests, parameters, and headers that are explicitly defined in your API specification, thereby ensuring security of your APIs by controlling the parameters that the backend receives. 

Reviewer: validations accept
```suggestion
Accept only those requests, parameters, and headers that are explicitly defined in your API specification, thereby ensuring security of your APIs by controlling the parameters that the backend receives. 
```

Reviewer: So why would anyone bother with "regular" validations?

Author: Done! Thanks!

== See Also

//* xref:https-reference.adoc[Configuring an HTTPS Endpoint]
* xref:download-proxy-task.adoc[Downloading an API Proxy]
* xref:proxy-latest-concept.adoc[Deploying an API Proxy]
* xref:proxy-deploy-cloudhub-latest-task.adoc[Deploying an API Proxy to CloudHub]
* xref:proxy-deploy-hybrid-latest-task.adoc[Deploying an API Proxy to Hybrid]

> * xref:wsdl-raml-http-proxy-reference.adoc[Handling API Proxy status codes]

Reviewer: ```suggestion
* xref:wsdl-raml-http-proxy-reference.adoc[Handling API Proxy Status Codes]
```

Author: Done! Thanks!
